sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,552 @@
+    authorizer: Option<Arc<Authorizer>>,
+}
+
+impl ServiceBusClient {
+    /// Creates a helper function to build AmqpConnectionOptions from ServiceBusClientOptions.
+    fn build_connection_options(
+        options: &ServiceBusClientOptions,
+    ) -> Option<AmqpConnectionOptions> {
+        if let Some(application_id) = &options.application_id {
+            let mut properties = AmqpOrderedMap::new();
+            properties.insert(
+                AmqpSymbol::from("user-agent"),
+                AmqpValue::from(application_id.clone()),
+            );
+            Some(AmqpConnectionOptions {
+                properties: Some(properties),
+                ..Default::default()
+            })
+        } else {
+            None
+        }
+    }
+    /// Returns a builder which can be used to create a new instance of [`ServiceBusClient`].
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_messaging_servicebus::ServiceBusClient;
+    /// use azure_identity::DeveloperToolsCredential;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     let credential = DeveloperToolsCredential::new(None)?;
+    ///     let client = ServiceBusClient::builder()
+    ///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn builder() -> ServiceBusClientBuilder {
+        ServiceBusClientBuilder::new()
+    }
+
+    /// Creates a new client internally using the provided options.
+    pub(crate) async fn new_internal(
+        fully_qualified_namespace: &str,
+        credential: Arc<dyn TokenCredential>,
+        options: ServiceBusClientOptions,
+    ) -> Result<Self> {
+        let endpoint = format!("amqps://{}:5671", fully_qualified_namespace);
+        let endpoint_url = Url::parse(&endpoint).map_err(|e| {
+            ServiceBusError::new(
+                ErrorKind::InvalidRequest,
+                format!("Invalid endpoint URL: {}", e),
+            )
+        })?;
+        let namespace = fully_qualified_namespace.to_string();
+
+        let connection = Arc::new(AmqpConnection::new());
+
+        // Create authorizer with the credential for token-based authentication
+        let authorizer = Arc::new(Authorizer::new(Arc::downgrade(&connection), credential));
+
+        let connection_options = Self::build_connection_options(&options);
+        connection
+            .open(
+                "servicebus-client".to_string(),
+                endpoint_url,
+                connection_options,
+            )
+            .await?;
+
+        Ok(Self {
+            connection,
+            namespace,
+            options,
+            authorizer: Some(authorizer),
+        })
+    }
+
+    /// Creates a sender for the specified queue or topic.
+    pub async fn create_sender(
+        &self,
+        queue_or_topic_name: &str,
+        _options: Option<CreateSenderOptions>,
```

Comment:
If you're getting `dead_code` or `unused_variables` on parameters, it means your public methods aren't being exported publicly correctly. This is a warning side you shouldn't just suppress by prefacing it with "_".